Sources/OfflineEntitlements/PurchasedProductsFetcher.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PurchasedProductsFetcher.swift
+//
+//  Created by Andrés Boedo on 3/17/23.
+
+import Foundation
+import StoreKit
+
+/// This struct should have all the information we need from StoreKit
+/// to create EntitlementInfo from a StoreKit 2 transaction.
+/// Other fields from other places might be needed.
+@available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+struct PurchasedSK2Product {
+
+    let productIdentifier: String
+    let periodType: PeriodType
+    let isActive: Bool
+    let willRenew: Bool
+    let latestPurchaseDate: Date?
+    let originalPurchaseDate: Date?
+    let expirationDate: Date?
+    let store = Store.appStore
+    let isSandbox: Bool
+    let unsubscribeDetectedAt: Date?
+    let billingIssueDetectedAt: Date?
+    let ownershipType: PurchaseOwnershipType
+    let verification: VerificationResult = .verified
+
+}
+
+@available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+extension PurchasedSK2Product {
```

Comment:
I think we need a different representation here, so we can use the same code that constructs `EntitlementInfos` from the response data. This is useful for creating an individual `EntitlementInfo`, but we'd have to change a bunch more code. Instead, we can use the types in `CustomerInfoResponse` so that at the top level we can construct a `CustomerInfo` the same way.

Might not make a lot of sense, but I'll have a PR ready to show what I mean.

Reply:
My new suggested type:
```swift
struct PurchasedSK2Product {

    let subscription: CustomerInfoResponse.Subscription
    let transaction: CustomerInfoResponse.Transaction
    let entitlement: CustomerInfoResponse.Entitlement

}
```

Reply:
#2358